### ISSUE TITLE
Out of the box Snap support

### ISSUE THREAD

**Issue by Hebilicious** [WhitewaterFoundry/Pengwin#618]
**Is your feature request related to a problem? Please describe.**
[snap](https://snapcraft.io/) requires systemctl wich isn't working yet in wsl ([wsl issue](https://github.com/microsoft/WSL/issues/5126) here).

**Describe the solution you'd like**
A more wsl-y solution will hopefully be provided my the wsl team soon. In the meantime it would be nice to have a workaround implemented in Pengwin.

**Describe alternatives you've considered**
There's some workarounds, check the WSL issues. An out of the box solution would be ideal.

**Additional context**
//

**Comment by crramirez** [comment]
Thank you for your suggestion we are working on something. But for Fedora Remix, so now we know that Pengwin users would like to have it also

**Comment by Hebilicious** [comment]
@crramirez Thanks for the quick reply, do you have a rough timeline ?

**Comment by crramirez** [comment]
No more than the end of the year, probably in two months

**Comment by crramirez** [comment]
Hello,

I recommend you to assist WSLConf. Will be a talk about snaps in WSL

Regards

**Comment by Peredery** [comment]
Is it still in WIP?

**Comment by crramirez** [comment]
Hello @Peredery,

Yes, we are still working on that but it will take more time to release a beta.

Regards,
Carlos

**Comment by kotx** [comment]
Any update on this?

**Comment by crramirez** [comment]
Hello @kotx,

Sadly no. But the fact that you ask for it, raises its priority.

Regards,
Carlos

**Comment by crramirez** [comment]
Hello @kotx @Hebilicious @Peredery 

The lastest version of Pengwin 22.11.4 includes an implementation of SystemD regardless you are in Windows 10 or 11. It can enabled via pengwin-setup

**Comment by flexchar** [comment]
I've tried installing snap using `sudo apt install snapd`. It went through. Then I wanted to `snap install hugo` but I get:

```
error: cannot communicate with server: Post "http://localhost/v2/snaps/hugo": dial unix /run/snapd.socket: connect: no such file or directory
```

I'm on
```
> lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Pengwin
Release:        11
Codename:       bullseye
```

Could someone clarify how I could use this feature? :)

**Comment by crramirez** [comment]
Hello @flexchar

Did you enabled systemd via pengwin-setup?

Regards

**Comment by flexchar** [comment]
Yep, I've just done it. And tried reinstalling snap but I'm sitting on the same error.

I can see it tried to create service but then it's not found.



```bash
lukas in ~ λ sudo apt install snapd
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Suggested packages:
  zenity | kdialog
The following NEW packages will be installed:
  snapd
0 upgraded, 1 newly installed, 0 to remove and 1 not upgraded.
Need to get 13.3 MB of archives.
After this operation, 56.5 MB of additional disk space will be used.
Get:1 https://deb.debian.org/debian bullseye/main amd64 snapd amd64 2.49-1+deb11u2 [13.3 MB]
Fetched 13.3 MB in 12s (1,155 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package snapd.
(Reading database ... 130377 files and directories currently installed.)
Preparing to unpack .../snapd_2.49-1+deb11u2_amd64.deb ...
Unpacking snapd (2.49-1+deb11u2) ...
Setting up snapd (2.49-1+deb11u2) ...
Created symlink /etc/systemd/system/multi-user.target.wants/snapd.recovery-chooser-trigger.service → /lib/systemd/system/snapd.recovery-chooser-trigger.service.
Created symlink /etc/systemd/system/multi-user.target.wants/snapd.seeded.service → /lib/systemd/system/snapd.seeded.service.
Created symlink /etc/systemd/system/cloud-final.service.wants/snapd.seeded.service → /lib/systemd/system/snapd.seeded.service.
Unit /lib/systemd/system/snapd.seeded.service is added as a dependency to a non-existent unit cloud-final.service.
Created symlink /etc/systemd/system/multi-user.target.wants/snapd.service → /lib/systemd/system/snapd.service.
Created symlink /etc/systemd/system/sockets.target.wants/snapd.socket → /lib/systemd/system/snapd.socket.
Processing triggers for man-db (2.10.1-1~bpo11+1) ...
Processing triggers for dbus (1.12.24-0+deb11u1) ...
Processing triggers for desktop-file-utils (0.26-1) ...
lukas in ~ λ service snap status
snap: unrecognized service
lukas in ~ λ service snapd status
snapd: unrecognized service
lukas in ~ λ service --status-all
 [ - ]  apparmor
 [ - ]  dbus
 [ ? ]  hwclock.sh
 [ - ]  procps
 [ - ]  rsync
 [ - ]  ssh
 [ - ]  sudo
 [ + ]  udev
 [ - ]  unattended-upgrades
 [ - ]  x11-common
 ```

**Comment by crramirez** [comment]
sudo systemctl enable --now snapd

**Comment by flexchar** [comment]
```bash
sudo systemctl enable --now snapd
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
```



I realized during `pengwin-setup` I get this:
![image](https://user-images.githubusercontent.com/9198896/210286120-7ddc4fac-a3c7-4530-acae-7d6bd55dc051.png)

```SystemD is only available in WSL2
If you want to start services in WSL1 you can use the "service" or the "wslsystemctl" commands.```

that may give any clue.


Okay, I also ran wsl version:
```bash
lukas in ~ λ wsl.exe -v
WSL version: 1.0.3.0
Kernel version: 5.15.79.1
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.963
```
I strongly believe I installed WSL 2 and I know for a fact that I'm running Win 11. I'm rather confused how these values exist since I've done a fresh WSL install after upgrading to Win 11... I guess I got some debug to do here

**Comment by crramirez** [comment]
Somethung is completely wrong in your pengwin installation that the Interop is not running.

Anyway, to enable systemd add

```
[boot]
systemd=true
```

To your /etc/wsl.conf file then `wsl --terminate WLinux`

And try again to start snapd

**Comment by flexchar** [comment]
Damn, given your support, you should totally charge more for this distro. I appreciate it!!

That being said, still not going through. I've also shut off docker in case it was messing up and ran `wsl --shutdown`. 

```bash
> sudo systemctl enable --now snapd
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
```

Meanwhile, to be respectful of your time, perhaps I should just dedicate time to wipe it all and do a fresh install. ;)

**Comment by crramirez** [comment]
You are very welcome! :-D

Considering the WSL version you sent, the systemd should already be working.

I you haven't wiped it all, could you please send me the content of your /etc/wsl.conf ?

Also, the content of $HOME/winhome/.wslconfig

**Comment by flexchar** [comment]
Absolutely!

```bash
lukas in ~ λ cat $HOME/winhome/.wslconfig
[wsl2]
memory=25GB
swap=0
localhostForwarding=true

[boot]
systemd=true
```

```bash
lukas in ~ λ cat /etc/wsl.conf
[automount]
enabled = true
options = "metadata,uid=1000,gid=1000,umask=0022,fmask=11,case=off"
mountFsTab = true
crossDistro = true

[network]
generateHosts = true
generateResolvConf = true

[filesystem]
umask = 0022

[user]
default=lukas
```

**Comment by crramirez** [comment]
Ok remove

```
[boot]
systemd=true
```

from `$HOME/winhome/.wslconfig` and add it to `/etc/wsl.conf`. Then terminate the distro, you should have systemd now.

I still don't see signals why pengwin-setup thinks that you are in WSL1. Could you please `echo $WSL_INTEROP`?

**Comment by flexchar** [comment]
It was the same. But since my vs code began hanging I decided to restart PC. It happens couple times a month. And after a reboot it worked. So it may have had worked all along if I've tried restarting it haha. 

But unfortunately, it wasn't without casualties. Docker back-end appears to be broken now. And tells that docker is not found in this WSL distro. Trying to re-enable it...

Oh, and I'll share all env variables with you:
```bash
lukas in ~ λ env | grep WSL
WSL2_GUI_APPS_ENABLED=1
WSLENV=WT_SESSION::WT_PROFILE_ID
WSL_INTEROP=/run/WSL/269_interop
WSL_DISTRO_NAME=WLinux
WSL_HOST_IP=172.19.224.1
```
